transport/src/main/java/io/scalecube/transport/TransportImpl.java

```diff
@@ -47,7 +47,7 @@
 
   private final TransportConfig config;
 
-  private final Subject<Message, Message> incomingMessagesSubject = PublishSubject.<Message>create().toSerialized();
+  private final Processor<Message, Message> incomingMessagesSubject = EmitterProcessor.create();
```

Comment:
FluxProcessor again instead of Processor. EmitterProcessor internally has SPSCQueue((https://github.com/JCTools/JCTools)) so serialized access will be guaranteed. However, I would recommend relying on the `FluxSink<>` functionality for manual element sending